daml-assistant/daml-helper/src/DA/Daml/Helper/Ledger.hs

```diff
@@ -345,6 +347,63 @@ allocateParty args name = do
         setRequestBodyJSON $ AllocatePartyRequest {identifierHint = text, displayName = text}
   return party
 
+runLedgerReset :: LedgerFlags -> IO ()
+runLedgerReset flags = do
+  putStrLn "Resetting ledger."
+  args <- getDefaultArgs flags
+  reset args
+
+reset :: LedgerArgs -> IO ()
+reset args =
+  case api args of
+    Grpc ->
+      runWithLedgerArgs args $ do
+        parties <- map L.party <$> L.listKnownParties
+        unless (null parties) $ do
+          ledgerId <- L.getLedgerIdentity
+          L.setToken
```

Comment:
Why do you need a token here? I see two cases:

1. Snackbox is running without auth. In that case, we don’t need one.
2. Snackbox is running with auth. In that case, the arbitrary secret used in `tokenFor` is not going to work unless sadbonx was started with `--auth-jwt-hs256-unsafe=secret`. At that point, not supporting auth seems easier to understand than supporting exactly one secret.

Note that this only applies to grpc, the JSON API needs the token.

Reply:
It's not needed, removed it.